ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/LondonGasCalculator.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import org.hyperledger.besu.ethereum.core.Account;
+import org.hyperledger.besu.ethereum.core.Gas;
+
+import org.apache.tuweni.units.bigints.UInt256;
+
+public class LondonGasCalculator extends BerlinGasCalculator {
+
+  // redefinitions for EIP-3529
+  private static final Gas SSTORE_CLEARS_SCHEDULE = SSTORE_RESET_GAS.plus(ACCESS_LIST_STORAGE_COST);
+
+  protected static final Gas NEGATIVE_SSTORE_CLEARS_SCHEDULE =
+      Gas.ZERO.minus(SSTORE_CLEARS_SCHEDULE);
+
+  // redefinitions for EIP-3529
+  private static final int NEW_MAX_REFUND_QUOTIENT = 5;
+
+  protected LondonGasCalculator() {}
+
+  // Redefined refund amount from EIP-3529
+  @Override
+  public Gas getSelfDestructRefundAmount() {
+    return Gas.ZERO;
+  }
+
+  // defined in Berlin, but re-implemented with new constants
+  @Override
+  // As per https://eips.ethereum.org/EIPS/eip-3529
+  public Gas calculateStorageRefundAmount(
+      final Account account, final UInt256 key, final UInt256 newValue) {
```

Comment:
this is identical to the berlin gas calculator, how about just accessors for the 2 changed constants?  The method is kinda funky to read, that is the only reason I ask

Reply:
At first I wanted to do this but it seems that this is what was also done for Berlin/Istanbul. Can be to improve readability or explained that this method has been updated. To maintain a consistency I wanted to keep that but we can in a future PR change that if you find it really necessary

Reply:
At first I wanted to do this but it seems that this is what was also done for Berlin/Istanbul. Can be to improve readability or explained that this method has been updated. To maintain a consistency I wanted to keep that but we can in a future PR change that if you find it really necessary